milavision/milavision/_utils.py

```diff
@@ -0,0 +1,53 @@
+from torch.utils.data import Dataset
+import socket
+import os
+import enum
+from pathlib import Path
+from typing import TypeVar
+
+from torchvision.datasets.vision import VisionDataset
+
+try:
+    pass
+except ImportError:
+    pass
+
+D = TypeVar("D", bound=Dataset)
+VD = TypeVar("VD", bound=VisionDataset)
+
+
+class ClusterType(enum.Enum):
+    """ Enum of the different clusters available. """
+
+    LOCAL = enum.auto()
+    MILA = enum.auto()
+    CEDAR = enum.auto()
+    BELUGA = enum.auto()
+    GRAHAM = enum.auto()
```

Comment:
I'm not sure why you're using an enum instead of something like

```python
if socket.getfqdn().endswith(".server.mila.quebec"):
    from . import mila as current
elif socket.getfqdn().endswith(".computecanada.ca"):  # ?
    from . import cc as current
...
```

I am going to note one thing here, which is that if this package is successful, a natural evolution of the idea is that other institutions than Mila may be interested in adding support for their own clusters (the enum might become quite large). That would be good, because if external code starts using this library for their own reasons, that is more code that plays nice on our cluster. This is one reason why I think that organizing code to group all code for a particular cluster together is a smart play.